integrations/terraform-provider-github
# Add support for repository alphanumeric autolinks (Fixes #1270)

### Changes

- Fixes #1270
- Add support for `is_alphanumeric` property of autolinks (available per https://github.com/google/go-github/pull/2450)

### Tests

- `github/resource_github_repository_autolink_reference_test.go`
- Adds tests when new `is_alphanumeric` property is not set (uses default) or is set to either `true` or `false`
- Replace usage of "OOF" with "TEST" to be more explicit that these are test resources.
- Update docs on usage

### How to Test
Create a personal access token with the following scopes: `repo`, `admin:org`, and `delete_repo`.

```shell
export GITHUB_TOKEN="ghp_xxx"
export GITHUB_ORGANIZATION="your-org"
TF_ACC=1 go test -v -run TestAccGithubRepositoryAutolinkReference
```

<details>
  <summary>Test Output</summary>

```
=== RUN   TestAccGithubRepositoryAutolinkReference
=== RUN   TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error
=== RUN   TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_anonymous_account
    resource_github_repository_autolink_reference_test.go:94: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_organization_account
=== RUN   TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error
=== RUN   TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_anonymous_account
    resource_github_repository_autolink_reference_test.go:207: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_organization_account
=== RUN   TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error
=== RUN   TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_anonymous_account
    resource_github_repository_autolink_reference_test.go:250: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_organization_account
--- PASS: TestAccGithubRepositoryAutolinkReference (60.26s)
    --- PASS: TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error (26.58s)
        --- SKIP: TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryAutolinkReference/creates_repository_autolink_reference_without_error/with_an_organization_account (26.58s)
    --- PASS: TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error (30.42s)
        --- SKIP: TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryAutolinkReference/imports_repository_autolink_reference_without_error/with_an_organization_account (30.42s)
    --- PASS: TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error (3.26s)
        --- SKIP: TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryAutolinkReference/deletes_repository_autolink_reference_without_error/with_an_organization_account (3.26s)
PASS
ok      github.com/integrations/terraform-provider-github/v5/github     60.741s
```
</details>

### Usage

Example configuration:
```tf
resource "github_repository_autolink_reference" "autolink" {
    repository          = github_repository.some-repo.name
    key_prefix          = "TICKET-"
    target_url_template = "https://example.com/TICKET-<num>"
    is_alphanumeric     = true
}
```
